turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -1336,22 +1492,42 @@ impl<B: Backend + 'static> TurboTasksApi for TurboTasks<B> {
 
     fn try_read_local_output(
         &self,
-        _parent_task_id: TaskId,
-        _local_task_id: LocalTaskId,
-        _consistency: ReadConsistency,
+        parent_task_id: TaskId,
+        local_task_id: LocalTaskId,
+        consistency: ReadConsistency,
     ) -> Result<Result<RawVc, EventListener>> {
-        todo!("bgw: local outputs");
+        // we don't currently support reading a local output outside of it's own task, so
+        // tracked/untracked is currently irrelevant
+        self.try_read_local_output_untracked(parent_task_id, local_task_id, consistency)
     }
 
     /// INVALIDATION: Be careful with this, it will not track dependencies, so
     /// using it could break cache invalidation.
     fn try_read_local_output_untracked(
         &self,
-        _parent_task_id: TaskId,
-        _local_task_id: LocalTaskId,
+        parent_task_id: TaskId,
+        local_task_id: LocalTaskId,
+        // we don't currently support reading a local output outside of it's own task, so
+        // consistency is currently irrelevant
         _consistency: ReadConsistency,
     ) -> Result<Result<RawVc, EventListener>> {
-        todo!("bgw: local outputs");
+        CURRENT_GLOBAL_TASK_STATE.with(|gts| loop {
+            let gts_read = gts.read().unwrap();
+            gts_read.assert_task_id(parent_task_id);
+            match gts_read.get_local_task(local_task_id) {
+                LocalTask::Unscheduled(..) => {
+                    drop(gts_read);
+                    self.schedule_local_task(parent_task_id, local_task_id);
```

Comment:
It's scheduled on read? That's different compared to how global functions work, they are scheduled on call.